gestalt/vault.py

```diff
@@ -1,29 +1,55 @@
+import os
 from datetime import datetime, timedelta
+from queue import Queue
+from threading import Thread
 from time import sleep
-from gestalt.provider import Provider
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import hvac  # type: ignore
 import requests
-from requests.exceptions import Timeout
 from jsonpath_ng import parse  # type: ignore
-from typing import Optional, Tuple, Any, Dict, Union, List
-import hvac  # type: ignore
-from queue import Queue
-import os
-from threading import Thread
-from retry import retry
+from requests.exceptions import Timeout
+from retry.api import retry_call
+
+from gestalt.provider import Provider
 
 
 class Vault(Provider):
-    @retry((RuntimeError, Timeout), delay=2, tries=5)  # type: ignore
-    def __init__(self,
-                 cert: Optional[Tuple[str, str]] = None,
-                 role: Optional[str] = None,
-                 jwt: Optional[str] = None,
-                 url: Optional[str] = os.environ.get("VAULT_ADDR"),
-                 token: Optional[str] = os.environ.get("VAULT_TOKEN"),
-                 verify: Optional[bool] = True,
-                 scheme: str = "ref+vault://") -> None:
+    def __init__(
+        self,
+        cert: Optional[Tuple[str, str]] = None,
+        role: Optional[str] = None,
+        jwt: Optional[str] = None,
+        url: Optional[str] = os.environ.get("VAULT_ADDR"),
+        token: Optional[str] = os.environ.get("VAULT_TOKEN"),
+        verify: Optional[bool] = True,
+        scheme: str = "ref+vault://",
+        delay: int = 2,
+        tries: int = 5,
+    ) -> None:
+        self.delay = delay
+        self.tries = tries
+
+        retry_call(
+            f=Vault.__do_init,
+            fargs=[self, cert, role, jwt, url, token, verify, scheme],
+            exceptions=(RuntimeError, Timeout),
+            delay=self.delay,
+            tries=self.tries,
+        )
+
+    # __init__ impl for retry_call
+    def __do_init(
+            self,
+            cert: Optional[Tuple[str, str]],
+            role: Optional[str],
+            jwt: Optional[str],
+            url: Optional[str],
+            token: Optional[str],
+            verify: Optional[bool],
+            scheme: str,
```

Comment:
i don't think this retry wrapper around the init method is necessary, and i think the original purpose of having a `retry` decorator around the init method in the first place was because we used to make vault requests during object initialization. 

we no longer make any requests during object initialization, so we are safe to remove any retry logic here.